src/snowflake/snowpark/dataframe.py

```diff
@@ -1359,31 +1359,24 @@ def select(
         if not exprs:
             raise ValueError("The input of select() cannot be empty")
 
-        # AST.
-        stmt = _ast_stmt
-        ast = None
-
-        if _emit_ast and _ast_stmt is None:
```

Comment:
why is this code moved below? Any reason? E.g., test failure?

Reply:
Yes, IDs were generated out of order because the stmt is created first and then ID's associated with any select expressions (in particular indirect table fn apply case) are created second.  However, the unparser resolves variables such that it expects any referenced variables (for example the table fn apply case) to have previously been defined and fails otherwise.

Reply:
sgtm, maybe let's add a comment to not forget this train of thought? Purposefully most of the code has been written so far in the form
```
if _emit_ast:
    ....

<snowpark code>

return <snowpark object>
```
This was with an eye towards removing the `<snowpark code>` / bringing it server-side. For that process, it may be easier to have a comment here to explain how this needs to be done here :)

Reply:
Added